Stratis.Bitcoin.Features.Consensus/StakeChainStore.cs

```diff
@@ -51,66 +49,58 @@ public StakeChainStore(Network network, ConcurrentChain chain, DBreezeCoinView d
 
             uint256 hash = await this.dBreezeCoinView.GetBlockHashAsync();
             ChainedBlock next = this.chain.GetBlock(hash);
-            var load = new List<StakeItem>();
 
+            var stakeItems = new List<StakeItem>();
             while (next != this.chain.Genesis)
             {
-                load.Add(new StakeItem { BlockId = next.HashBlock, Height = next.Height });
-                if ((load.Count >= this.threshold) || (next.Previous == null))
+                stakeItems.Add(StakeItem.Query(next.HashBlock, next.Height));
+                if ((stakeItems.Count >= this.threshold) || (next.Previous == null))
                     break;
                 next = next.Previous;
             }
 
-            await this.dBreezeCoinView.GetStake(load);
+            await this.dBreezeCoinView.GetStakeItems(stakeItems);
 
-            // All block stake items should be in store.
-            if (load.Any(l => l.BlockStake == null))
-            {
-                this.logger.LogTrace("(-)[STAKE_INFO_MISSING]");
-                throw new ConfigurationException("Missing stake information, delete the data folder and re-download the chain");
-            }
-
-            foreach (StakeItem stakeItem in load)
-                this.items.TryAdd(stakeItem.BlockId, stakeItem);
+            foreach (StakeItem stakeItem in stakeItems)
+                this.items.TryAdd(stakeItem.BlockHash, stakeItem);
```

Comment:
You removed the check that all items are not null why?
```
  if (load.Any(l => l.BlockStake == null))
```
If we did not have this check til now finding the bug in the first place would be much harder.